### DISCUSSION TITLE
Can a workflow save itself?

### DISCUSSION THREAD

**Discussion by ablot** [bonsai-rx/bonsai#1002] [Q&A] [answered]
Hi all,

Can a workflow write a copy of itself? I'm looking for something like a node that triggers "Save as" when it receives an input. Is that possible?

A bit more of context for why that would be useful. I use bonsai to control my experimental setup and record ephys/videos. The workflow I use evolves slowly with each iteration of the experiments. I try to save all relevant parameters but it is sometimes a bit hard to find the exact version of the workflow that was used. I can use "Save as" to keep a copy with the data to help but it will save only the main workflow (not the included) and does not save the parameters that are set at initialisation, or manually adjusted at the beginning of the recording. I was wondering if I could have a "save" node in each included workflow and trigger them all when I start to acquire data.
Obviously I can also copy my whole bonsai folder in the data directory before each experiment and run it directly from there. It makes it just a bit slower and harder to switch fast between experiments and protocols if i have to copy/paste instead of just press stop workflow, change one parameter, restart.

**Comment by bruno-f-cruz** [comment, marked as answer]
There is currently no one node that does this. I have attached a custom c# script that should be able to copy and paste a target file. In this case you would simply target the workflow itself.
Let me know if it works!

[saveworkflow.zip](https://github.com/bonsai-rx/bonsai/files/9299889/saveworkflow.zip)

**Reply by ablot** [reply]
It's great. It works nicely. I just need to hit Ctrl+S before starting the workflow (to save actually the version I run, not the previous one that is on disk). 
The only drawback is that it will not save the parameters that are dynamically updated at initialisation. But if it's the only thing I need to remember to explicitely save, that's fine.
